source/httpfs/http_filesystem_test.go

```diff
@@ -0,0 +1,170 @@
+//go:generate esc -o escfs/escfs.go -pkg escfs -prefix escfs/ escfs/
+package httpfs
+
+import (
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"os"
+	"path"
+	"path/filepath"
+	"testing"
+
+	"github.com/golang-migrate/migrate/v4/source/httpfs/escfs"
+	st "github.com/golang-migrate/migrate/v4/source/testing"
+)
+
+func Test(t *testing.T) {
+	tmpDir, err := ioutil.TempDir("", "")
```

Comment:
As mentioned [here](https://github.com/golang-migrate/migrate/pull/163#discussion_r252911552), don't use a a temp dir for testing. You can use `afero` instead.